### PR TITLE
[106] Clear session when clicking "Cancel" button in pipeline wizard

### LINKED ISSUE #106: "Cancel" button in add pipeline wizard should clear session values
A/C:

- [x] If someone leaves the form by accident, they should see their pending changes when they come back.
- [x] If they clicked the cancel button, it should erase those values from the session.
- [x] Nice to have: Display a message to the user "This form contains unsaved changes." when the page loads on a `GET` request and the form is populated with unsaved values from the session.

### PR DESCRIPTION
**Pull request by ronardcaktus**
Clears the session if the user clicks the `Cancel` button on the new-pipeline page (step 1). If the user exits without clicking `Cancel`, they should still see the form populated upon returning.

Closes #106

### CODE REVIEW

File: etlman/projects/views.py [hunk 1]
```diff
@@ -214,6 +221,13 @@ def clear_step_wizard_session_variables(request):
         request.session.pop(session_key.value, None)
 
 
+@authorize(user_is_project_collaborator)
```
Comment (tobiasmcnulty) [inline etlman/projects/views.py:243]: 💯

File: etlman/projects/tests/test_views/test_view_pipeline.py [hunk 2]
```diff
@@ -422,3 +422,17 @@ def test_back_edit_step_post(self, nonadmin_client, project):
         assert step_session_key in nonadmin_client.session
         assert nonadmin_client.session[step_session_key]["name"] == data["name"]
         assert pipeline.name in str(response.content)
+
+
+@pytest.mark.django_db
+class TestClearSession:
+    def test_cancel_new_step_get(self, nonadmin_client, project):
+        response = nonadmin_client.get(
```
Comment (tobiasmcnulty) [inline etlman/projects/tests/test_views/test_view_pipeline.py]: This test is not really testing the functionality of the view. Proposed improvements:

- [x] Call `save_step2_data_in_session()` to save some values to the session. Maybe started with unsaved (`.build()`) model objects (including `Pipeline`, `DataInterface`, and `Step`)
- [x] Call `nonadmin_client.get()` to execute the view
- [x] Inspect `nonadmin_client.session` to ensure it no longer contains the keys in question
- [x] Create a second test that follows the same pattern and works with model objects that have been saved to the DB (i.e., created with a factory _without_ the `.build()` suffix)

**Review by tobiasmcnulty — COMMENTED** [review]
I tested this in the browser and I am afraid it's not working for me; after clicking the cancel button and navigating back to the form, I still see my pending changes.

https://www.dropbox.com/s/qchgj726qz9d9u9/etlman-106-bug.mov?dl=0

I included some thoughts below about expanding the tests for this code which may help track down the issue.

Also:
- [x] `project.min.js`, `vendors.min.js`, and `project.css` need to be removed from git (they're in `.gitignore`, but still checked into the repo)

File: etlman/projects/tests/test_views/test_view_pipeline.py (hunk 2 above)
Comment (ronardcaktus) [reply etlman/projects/tests/test_views/test_view_pipeline.py]: I think all suggestions are addressed.

File: etlman/projects/views.py (hunk 1 above)
Comment (ronardcaktus) [reply etlman/projects/views.py:243]: If `vendor.min.js` is removed the branch stops working. How should we address that?
I removed the other two files.

**Review by tobiasmcnulty — APPROVED** [review]
Looks good and works for me! 🏅

**Pull request merged**